### DISCUSSION TITLE
Is there any way I can have both open and closed sourced version?

### DISCUSSION THREAD

**Discussion by TheRealOne78** [NVIDIA/open-gpu-kernel-modules#304] [Q&A] [answered]
Is there any way I can have both open and closed sourced version?

**Comment by TheTechRobo** [comment]
Why...?

**Comment by TheRealOne78** [comment]
For testing purposes.
Having 2 versions which you can switch between in the same session is faster and better than compiling, installing and having to reboot everytime.

**Comment by aritger** [comment, marked as answer]
Sorry, there isn't an official way to have both sets of kernel modules installed at the same time.  Both sets of kernel modules provide files with the same names (nvidia.ko, nvidia-modeset.ko, etc), and modprobe wouldn't know which set you want to use.

That said, you could probably be clever with a script that copies one set of kernel modules or the other into place, such that the script effectively "switches" between the two sets.  This would require some care though; I wouldn't recommend this if you're not pretty familiar with the files involved.

**Reply by TheRealOne78** [reply]
Thanks. I will try with that
